regression/cbmc/Pointer_difference2/main.c

```diff
@@ -0,0 +1,16 @@
+int main()
+{
+  int array[4];
+  int other_array[2];
+
+  __CPROVER_assert(&array[0] - &array[2] == -2, "correct");
+
```

Comment:
Would I be awful to request...

`__CPROVER_assert(((char *)(&array[3])) - ((char *)(&array[1])) == 2*sizeof(int), "casting works");`

and

```
int *p = &array[3];
++p;
__CPROVER_assert(p - &array[0] == 4, "end plus one works");
__CPROVER_assert(p - &array[0] != 3, "end plus one works");
++p;
_Bool nondet_branch;
if (nondet_branch)
  __CPROVER_assert(p - &array[0] == 5, "end plus 2 is nondet");
else
  __CPROVER_assert(p - &array[0] != 5, "end plus 2 is nondet");
```

Reply:
Done, thanks!